### PR TITLE
migrate part of painting to nullsafety

### PR DESCRIPTION
**Pull request by a14n**
## Description

NNBD migration for a part of painting

## Related Issues

* dart-lang/sdk#40146

## Tests

I added the following tests:

None

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/painting/box_decoration.dart [hunk 1]
```diff
@@ -213,18 +212,18 @@ class BoxDecoration extends Decoration {
   final BoxShape shape;
 
   @override
-  EdgeInsetsGeometry get padding => border?.dimensions;
+  EdgeInsetsGeometry? get padding => border?.dimensions;
 
   @override
-  Path getClipPath(Rect rect, TextDirection textDirection) {
-    Path clipPath;
+  Path? getClipPath(Rect rect, TextDirection textDirection) {
```
Comment (goderbauer) [inline packages/flutter/lib/src/painting/box_decoration.dart:218]: This is surprising, shape is never null, so this should always return non-null.

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/painting/box_decoration.dart (hunk 1 above)
Comment (a14n) [reply packages/flutter/lib/src/painting/box_decoration.dart:218]: null can be returned with BoxShape.rectangle and a null borderRadius.

**Comment by a14n** [comment]
>Google testing failed. Please contact a Google developer to investigate.

@goderbauer could you take a look please?

**Comment by goderbauer** [comment]
It looked like a flake. I have restarted it.

**Comment by a14n** [comment]
>It looked like a flake. I have restarted it.

Still red :-/

**Comment by goderbauer** [comment]
Ugh, it's failing because this PR depends on changes in https://github.com/flutter/flutter/commit/2c24e152e418f7c2e008fec7f5a662da9e3eb3a6 and those have not rolled into google3 yet. Let me check if we can just ignore the failure for now.

**Comment by goderbauer** [comment]
I restarted the internal test with the base commit of this PR included. Let's see how that goes...

**Comment by goderbauer** [comment]
Google testing is green now.

**Comment by alectogeek** [comment]
This upgrade causes errors because there were deleted "@" symbol before `@required`

**Comment by alectogeek** [comment]
@a14n probably it is just a typo? Or there is something else? Because I'm not the only person who has the problem #67084

**Comment by goderbauer** [comment]
@alectogeek required without the @ is the new syntax. If you restart your analyzer server, those problems should go away.

**Pull request merged**